xlconnect.Rmd

```diff
@@ -0,0 +1,544 @@
+```
+
+The data frame can also be written in a named region using the single call `writeNamedRegionToFile()` function. This will create the sheet (if necessary) and named region name based on the formula argument and call writeNamedRegion instead of writeWorksheet.
+```{r , echo = TRUE, eval = FALSE}
+# Write and save a data frame into a specific named region of a workbook
+writeNamedRegionToFile("xlconnect-demo/XLConnectDemo.xlsx", 
+                       data = swissfranc, name = "SwissFrancName", 
+                       formula = "NamedRegionEx!$B$2")
+```
+
+### Reading from  Excel files
+
+```{r , echo = TRUE, eval = FALSE}
+# Load Excel workbook - either .xls or .xlsx. Create if not existing.
+wb <- loadWorkbook("xlconnect-demo/XLConnectDemo.xlsx", create = TRUE)
+
+# Read data from a specific sheet of a workbook
+dataFromExcel <- readWorksheet(wb, sheet = "CHF")
+
+# Read data from a specific location in a sheet of a workbook
+dataFromExcel <- readWorksheet(wb, 
+                               sheet = "CHF", 
+                               startRow = 2, endRow = 10,
+                               startCol = 2, endCol = 5, 
+                               header = TRUE)
+```
+
+The steps above can be executed in a single call using the `readWorksheetFromFile()` function.
+```{r , echo = TRUE, eval = FALSE}
+# Read data from a specific location in a sheet of a workbook
+dataFromExcel <- readWorksheetFromFile("xlconnect-demo/XLConnectDemo.xlsx", 
+                                       sheet = "OneCall", 
+                                       startRow = 2, endRow = 10, 
+                                       startCol = 2, endCol = 5, 
+                                       header = TRUE)
+```
+
+The data can also be read from an existing named region using the single call `readNamedRegionFromFile()` function.
+```{r , echo = TRUE, eval = FALSE}
+# Read data from a specific named region of a workbook
+dataFromExcel <- readNamedRegionFromFile("xlconnect-demo/XLConnectDemo.xlsx", 
+                                         name = "SwissFrancName", 
+                                         header = TRUE)
+```
+
+It can be useful to retrieve all the sheets of a workbook or all existing named regions.
+```{r , echo = TRUE, eval = FALSE}
+# Get all sheet names of a workbook
+allSheets <- getSheets(wb)
+
+# Get all named regions of a workbook
+allnr <- getDefinedNames(wb)
+```
+
+## Formatting
+
+### Create and apply some formatting
+Now let's try out some formatting functionality.
+
+```{r , echo = TRUE, eval = FALSE}
+# If previous chunks have not been run, the file may not exist
+if(!exists("xlconnect-demo/XLConnectDemo.xlsx")) {
+    # Write and save a data frame into a specific named region of a workbook
+  writeNamedRegionToFile("xlconnect-demo/XLConnectDemo.xlsx", 
+                       data = swissfranc, name = "SwissFrancName", 
+                       formula = "NamedRegionEx!$B$2")
+} 
```

Comment:
```suggestion
```
I get an error here (running each cell above in a clean session):
Error: IllegalArgumentException (Java): Specified name 'SwissFrancName' already exists!
But also don't get the point of this block, it does the same as one of the calls above.